### ISSUE TITLE
[Enhancement]: Remove redundant `source` parameter from `aws_securitylake_aws_log_source`

### ISSUE THREAD

**Issue by gdavison** [hashicorp/terraform-provider-aws#37160]
### Description

The resource type `aws_securitylake_aws_log_source` has a single root parameter, `source`. This is because the AWS API request takes multiple `Sources. However, the design of the resource allows it to configure a single log source.

### Affected Resource(s) and/or Data Source(s)

aws_securitylake_aws_log_source

### Potential Terraform Configuration

```terraform
N/A
```

### References

N/A

### Would you like to implement a fix?

None

**Comment by jar-b** [comment]
We have not yet deprecated the `source` argument, so we need to begin a deprecation cycle before we can remove it in favor of root level arguments. This issue will be moved into `v7.0.0` and a new issue will be opened to track the deprecation effort.

**Comment by jar-b** [comment]
We've decided to defer the removal of the `source` block and conversion to root-level arguments for the time being. While the current schema design is a miss on our end - the `source` block is redundant as each resource is by design only one source - we recognize that fixing it comes with a requirement for users to migrate existing configurations to a new pattern.

Until automated upgrade tooling is available, the additional refactoring burden this would place on users does not outweigh the desire to fix a design mistake. This issue will remain in place to track the concept and community interest, but will not be placed in a major version milestone.
